otherlibs/unix/socketpair_win32.c

```diff
@@ -36,61 +37,30 @@ extern const int caml_unix_socket_type_table[]; /* from socket.c */
 
 static int socketpair(int domain, int type, int protocol,
                       SOCKET socket_vector[2],
-                      BOOL inherit)
+                      BOOL inherit, char *sun_path)
 {
-  wchar_t dirname[MAX_PATH + 1], path[MAX_PATH + 1];
-  union sock_addr_union addr;
-  socklen_param_type socklen;
-
   /* POSIX states that in case of error, the contents of socket_vector
      shall be unmodified. */
   SOCKET listener = INVALID_SOCKET,
     server = INVALID_SOCKET,
     client = INVALID_SOCKET;
 
-  fd_set writefds, exceptfds;
-  u_long non_block, peerid = 0UL;
-
-  DWORD drc;
+  bool defer_delete_file = false;
   int rc;
 
-  if (GetTempPath(MAX_PATH + 1, dirname) == 0) {
-    caml_win32_maperr(GetLastError());
-    goto fail;
-  }
-
-  if (GetTempFileName(dirname, L"osp", 0U, path) == 0) {
-    caml_win32_maperr(GetLastError());
-    goto fail;
-  }
-
-  addr.s_unix.sun_family = PF_UNIX;
-  socklen = sizeof(addr.s_unix);
-
-  /* sun_path needs to be set in UTF-8 */
-  rc = WideCharToMultiByte(CP_UTF8, 0, path, -1, addr.s_unix.sun_path,
-                           UNIX_PATH_MAX, NULL, NULL);
-  if (rc == 0) {
-    caml_win32_maperr(GetLastError());
-    goto fail_path;
-  }
-
   listener = caml_win32_socket(domain, type, protocol, NULL, inherit);
   if (listener == INVALID_SOCKET)
     goto fail_wsa;
 
-  /* The documentation requires removing the file before binding the socket. */
-  if (DeleteFile(path) == 0) {
-    drc = GetLastError();
-    if (drc != ERROR_FILE_NOT_FOUND) {
-      caml_win32_maperr(drc);
-      goto fail_sockets;
-    }
-  }
+  union sock_addr_union addr = { 0 };
+  socklen_param_type socklen = sizeof(addr.s_unix);
+  addr.s_unix.sun_family = PF_UNIX;
+  memcpy(addr.s_unix.sun_path, sun_path, strlen(sun_path));
```

Comment:
Nit: Even though I do see that you assert the length does not exceed `UNIX_PATH_MAX` in the caller, and, that the union is zeroed out at the declaration site, in isolation not using a `str{n|l}cpy` feels a bit brittle, imho.